### PR TITLE
Added support for Pydantic 2.x

### LINKED ISSUE #574: Support for pydantic 2.0
Currently only pydantic>=1.10.0,<2 is accepted in requirements.

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
Closes #574 

Added support for Pydantic 2.x. Support for Pydantic 1.x is retained.

Also ensure that the mapping example runs with the very old version of Pint (v0.16) required by AiiDA.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Test update

## Checklist for the reviewer
This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: examples/datamodel_as_rdf/dataresource.py [hunk 1]
```diff
@@ -85,7 +92,7 @@ class ResourceConfig(BaseModel):
         ),
     )
 
-    @root_validator
+    @root_validator(skip_on_failure=True)
```
Comment (francescalb) [inline examples/datamodel_as_rdf/dataresource.py:95]: I do not understand this, why should a test be skipped if it fails?

File: examples/datamodel_as_rdf/dataresource.py [hunk 2]
```diff
@@ -2,12 +2,19 @@
...
 
 import dlite
 from dlite.rdf import from_rdf, to_rdf
 from dlite.utils import pydantic_to_instance, pydantic_to_metadata
 
 
+# Require Pydantic V1
+if int(pydantic_version.split(".")[0]) != 1:
+    print("This example requires pydantic V1")
+    raise SystemExit(44)
```
Comment (francescalb) [inline examples/datamodel_as_rdf/dataresource.py:15]: Can you make a version for pydantic 2?

File: bindings/python/utils.py [hunk 3]
```diff
@@ -221,7 +220,8 @@ def get_dataclass_entity_schema():
     @dataclass
     class Property:
         type: str
-        dims: Optional[List[str]]
+        #ref: Optional[str]  # Hmm, how to create field name "@ref"?
```
[the remark this answers is not in the document]
Comment (jesper-friis) [reply bindings/python/utils.py:223]: Updated the comment and added issue #595

**Review by francescalb — APPROVED** [review]
Seems fine.

**Pull request merged**